aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -687,6 +692,250 @@ public void onError(Exception error) {
         );
     }
 
+    @Override
+    public void fetchUserAttributes(
+            @NonNull Consumer<List<AuthUserAttribute>> onSuccess,
+            @NonNull Consumer<AuthException> onError
+    ) {
+        awsMobileClient.getUserAttributes(new Callback<Map<String, String>>() {
+            @Override
+            public void onResult(Map<String, String> result) {
+                List<AuthUserAttribute> userAttributes = new ArrayList<>();
+                for (Map.Entry<String, String> entry : result.entrySet()) {
+                    userAttributes.add(new AuthUserAttribute(
+                            new AuthUserAttributeKey(entry.getKey()), entry.getValue()));
+                }
+                onSuccess.accept(userAttributes);
+            }
+
+            @Override
+            public void onError(Exception error) {
+                onError.accept(new AuthException(
+                        "Failed to fetch user attributes since user is not authenticated",
+                        error,
+                        "See attached exception for details"
+                ));
+            }
+        });
+    }
+
+    @Override
+    public void updateUserAttribute(
+            @NonNull AuthUserAttribute attribute,
+            @NonNull Consumer<AuthUpdateAttributeResult> onSuccess,
+            @NonNull Consumer<AuthException> onError
+    ) {
+
+        awsMobileClient.updateUserAttributes(
+                Collections.singletonMap(attribute.getKey().getKeyString(), attribute.getValue()),
+                new Callback<List<UserCodeDeliveryDetails>>() {
+                    @Override
+                    public void onResult(List<UserCodeDeliveryDetails> result) {
+                        if (!isValidUserAttribute(attribute.getKey().getKeyString())) {
```

Comment:
If the user reached this success callback, that means they sent an attribute to update which Cognito accepted and successfully updated. We certainly don't want to throw an error in that case. 

Also, Cognito supports custom attributes in addition to the standard ones we have listed so we don't know whether an attribute is valid or not - hence you can't have a working isValidUserAttribute method.